### PR TITLE
[addition] Virtual Midi Device Creation + midi devices bugfixes

### PR DESCRIPTION
**Pull request by RobinMontferme**
- Enabled creation of virtual Midi device from device explorer
- Fixed a bug where Devices weren't having their default settings applied
- Modified default settings for `MIDIInput` and `MIDIOutput` devices to set the correct port direction and added the new virtual parameter 
- Made so that the name of the device is also the name of the device in JACK
- made `MIDIInputProtocolFactory` and `MIDIOutputProtocolFactory` classes member public

### CODE REVIEW

File: src/plugins/score-plugin-protocols/Protocols/MIDI/MIDIProtocolSettingsWidget.cpp [hunk 1]
```diff
@@ -31,14 +31,20 @@ MIDIInputSettingsWidget::MIDIInputSettingsWidget(QWidget* parent)
...
   lay->addRow(tr("Name"), m_name);
   lay->addRow(m_createWhole);
+  lay->addRow(m_virtualPort);
 
   setLayout(lay);
   m_createWhole->setChecked(false);
-  m_createWhole->setEnabled(true);
+  m_createWhole->setEnabled(false);
+  m_virtualPort->setChecked(false);
+#if !__WIN32
```
Comment (jcelerier) [inline src/plugins/score-plugin-protocols/Protocols/MIDI/MIDIProtocolSettingsWidget.cpp]: I don't think this works as "enabled" is the default state, it should be 

```
#if defined(_WIN32) // note the single underscore
  m_virtualPort->setEnabled(false);
#endif
```

instead (with maybe a comment to say that this is because win32 does not support that)

File: src/plugins/score-plugin-protocols/Protocols/MIDI/MIDIProtocolSettingsWidget.cpp [hunk 2]
```diff
@@ -83,14 +93,20 @@ MIDIOutputSettingsWidget::MIDIOutputSettingsWidget(QWidget* parent)
...
   auto lay = new QFormLayout;
   lay->addRow(tr("Name"), m_name);
   lay->addRow(m_createWhole);
+  lay->addRow(m_virtualPort);
 
   setLayout(lay);
   m_createWhole->setChecked(false);
   m_createWhole->setEnabled(true);
+  m_virtualPort->setChecked(false);
+#if __WIN32
```
Comment (jcelerier) [inline src/plugins/score-plugin-protocols/Protocols/MIDI/MIDIProtocolSettingsWidget.cpp]: Same

File: src/plugins/score-plugin-protocols/Protocols/MIDI/MIDISpecificSettingsSerialization.cpp [hunk 3]
```diff
@@ -44,4 +45,8 @@ void JSONWriter::write(Protocols::MIDISpecificSettings& n)
     n.api = (libremidi::API)it->toInt();
   else
     n.api = libremidi::default_platform_api();
+  if (auto it = obj.tryGet("VirtualPort"))
```
Comment (jcelerier) [inline src/plugins/score-plugin-protocols/Protocols/MIDI/MIDISpecificSettingsSerialization.cpp:48]: perfect thanks

File: src/plugins/score-plugin-protocols/Protocols/MIDI/MIDIProtocolSettingsWidget.cpp [hunk 4]
```diff
@@ -120,13 +143,6 @@ void MIDIOutputSettingsWidget::setSettings(
     ossia::net::sanitize_device_name(prettyName);
   }
   m_name->setText(prettyName);
-
-  if (settings.deviceSpecificSettings.canConvert<MIDISpecificSettings>())
```
Comment (jcelerier) [inline src/plugins/score-plugin-protocols/Protocols/MIDI/MIDIProtocolSettingsWidget.cpp:124]: hmmm why was this part removed ?

**Review by jcelerier — CHANGES_REQUESTED** [review]

**Comment by smoothdeveloper** [comment]
🎉

File: src/plugins/score-plugin-protocols/Protocols/MIDI/MIDIProtocolSettingsWidget.cpp (hunk 1 above)
Comment (RobinMontferme) [reply src/plugins/score-plugin-protocols/Protocols/MIDI/MIDIProtocolSettingsWidget.cpp]: Done

**Review by jcelerier — APPROVED** [review]
:+1:

**Comment by jcelerier** [comment]
erf there's a conflict with libossia because of the merge of the other commit. Can you : 

- go into master
- git pull; git submodule update --init --recursive
- go into your branch
- git add 3rdparty/libossia # it should set it to the latest commit
- git commit --amend
- git push --force

and we'll be able to merge
